packages/framework/src/evaluate.ts

```diff
@@ -63,8 +73,41 @@ export const evaluate = (
   context?: Record<string, unknown>,
 ): unknown => {
   try {
-    const fn = buildEvaluateFn(screen, js, context);
-    return fn();
+    const expressions = countBy(compact(js?.split("${")), (i) =>
+      i.startsWith("${"),
+    );
+
+    if (js === undefined || toLength(values(expressions)) <= 1)
+      return buildEvaluateFn(screen, js, context)();
+    else {
+      let result = js;
+
+      let searchIndex = 0;
+      while (true) {
+        const open = result.indexOf("${", searchIndex);
+        if (open === -1) break;
+
+        const close = result.indexOf("}", open);
+        if (close === -1) break;
+
+        const template = result.slice(open + 2, close);
+        const value = buildEvaluateFn(screen, template, context)();
+
+        let replaced = value;
+        if (isArray(value)) replaced = `[${value}]`;
+        else if (typeof value === "object") replaced = JSON.stringify(value);
+
+        result = `${result.slice(
+          0,
+          open - (isArray(value) ? 1 : 0),
+        )}${replaced}${result.slice(close + 1 + (isArray(value) ? 1 : 0))}`;
+
+        searchIndex = open + toLength(value);
+      }
+      isString(result) && (result = JSON.parse(result));
+
+      return result;
+    }
```

Comment:
I think this is overcomplicating things. If the value of js is a multiline string, then we can just evaluate the entire block, no need to evaluate the expressions individually.

The main problem we have to solve is whether you return an object or an interpolated string and expect it to get parsed later on. We might need an extra param for that since I'm not sure there's a good heuristic to determine what to return from the JS alone. Maybe just checking the object braces is enough....